### ISSUE TITLE
MessageResolver throws a NoSuchMessageException if a message cannot be resolved (and does not return null as mentioned in the javadoc)

### ISSUE THREAD

**Issue by ingogriebsch** [spring-projects/spring-hateoas#1380]
If one calls `resolve` on the `MessageResolver` and the message cannot be resolved, a `NoSuchMessageException` is thrown. 

The javadoc of the `resolve` method explains that `null` is returned if no message was found. This is only true if the no resource bundles were found and the `DefaultOnlyMessageResolver` is configured through the `HateoasConfiguration`.

**Comment by odrotbohm** [comment]
Are you fine with the current behavior and just want us to update the Javadoc? Or are you rather suggesting we should change the behavior?

**Comment by ingogriebsch** [comment]
I'm personally a bit unsure about how to handle this issue. If you not have asked I would have said, that the behavior needs to be changed for sure.

I just created a test project to check why the exception is not thrown if somone is for example using HAL as a mediatype. And I realized that the reason is, that class `HalLinkRelation` (which is a `MessageSourceResolvable`) returns an empty string if something calls `getDefaultMessage` on it.

My implementation of `MessageSourceResolvable` is simply returning `null` (because it does not know about a good/matching default message). Therefore the underlying `MessageSource` can neither resovle a message nor a default message and therefore throws an exception.

I'm currently solving the problem the way, that I `try/catch` the call to the given `MessageResolver`. I could also return an empty String on all my `MessageSourceResolvable` implementations (but this feels like it is not the correct way to do).

tldr: I'm still unsure. Currently I think 'Why do I need to handle that case? Especially because there is a facility available in Spring HATEOAS that deals with the message-source/resource-bundle.'. But I can live/deal with weither option you have in mind as a solution! :)
